qiita_ware/processing_pipeline.py

```diff
@@ -232,27 +231,54 @@ def _get_preprocess_fasta_cmd(raw_data, prep_template, params):
             quals.append(join(output_dir, '%s.qual' % base))
         prepreprocess_cmd = '; '.join(prepreprocess_cmds)
 
-    if quals:
-        qual_str = "-q %s -d" % ','.join(quals)
-
     # The minimal QIIME mapping files should be written to a directory,
     # so QIIME can consume them
     prep_dir = mkdtemp(dir=qiita_config.working_dir,
                        prefix='MMF_%s' % prep_template.id)
 
     # Get the Minimal Mapping Files
-    mapping_fp = _get_qiime_minimal_mapping(prep_template, prep_dir)[0]
+    mapping_fps = sorted(_get_qiime_minimal_mapping(prep_template, prep_dir))
 
     # Add any other parameter needed to split libraries
     params_str = params.to_str()
 
     # Create the split_libraries_fastq.py command
-    cmd = ' '.join(["split_libraries.py",
-                    "-f %s" % ','.join(seqs),
-                    "-m %s" % mapping_fp,
-                    qual_str,
-                    "-o %s" % output_dir,
-                    params_str])
+    # len(mapping_fps) will be == 1 when there is no run_prefix or it has
+    # the same value in all the rows
+    if len(mapping_fps) == 1:
+        qual_str = "-q %s -d" % ','.join(quals) if quals else ""
+        cmd = ' '.join(["split_libraries.py",
+                        "-f %s" % ','.join(seqs),
+                        "-m %s" % mapping_fps[0],
+                        qual_str,
+                        "-o %s" % output_dir,
+                        params_str])
+    else:
+        cmd, output_folders, n = [], [], 1
+        for i, (seq, mapping) in enumerate(zip(seqs, mapping_fps)):
+            qual_str = "-q %s -d" % quals[i] if quals else ""
+            split_dir = join(output_dir, basename(mapping))
+            output_folders.append(split_dir)
+
+            cmd.append(' '.join(["split_libraries.py",
+                                 "-f %s" % seq,
+                                 "-m %s" % mapping,
+                                 qual_str,
+                                 "-o %s" % split_dir,
+                                 "-n %d" % n,
+                                 params_str]))
+            # Number comes from (100K larger than amplicon):
+            # http://454.com/products/gs-FLX-system/index.asp
+            n = (i + 1) * 800000
+
+        # files to cat from multiple split libraries
+        to_cat = ['split_library_log.txt', 'seqs.fna', 'seqs_filtered.qual']
+        for tc in to_cat:
+            files = map(lambda x: join(x, tc), output_folders)
```

Comment:
I think it has to do with some py3 incompatibilities: in py3, map its a generator, while in py2 it actually returns the list. Agree in minor/non-blocking, so it`s up to @antgonza if he wants to change it.